exercises/practice/space-age/.approaches/introduction.md

```diff
@@ -0,0 +1,117 @@
+
+Suppose you are on Mercury.
+The [orbital period][wikipedia-orbital-period] of Mercury is approximately one fourth of an Earth year.
+This means that in one Earth year, Mercury will complete about four whole orbits around the sun.
+As a consequence, your 'age' on Mercury is about four times your age on Earth.
+
+More generally, your 'age' on a certain planet is the number of seconds since your birth, divided by the number of seconds it takes the planet to orbit the sun once.
+
+The various planets' orbital periods as compared to the Earth's are given.
+To convert them from _Earth years_ per orbit to _seconds_ per orbit you need to multiply this by the number of seconds in one Earth year.
+
+
+## Approach: give meaningful names to important values
+
+```haskell
+ageOn :: Planet -> Float -> Float
+ageOn planet seconds = seconds / (periodInEarthYears * secondsPerEarthYear)
+  where
+    secondsPerEarthYear = 60 * 60 * 24 * 365.25
+
+    periodInEarthYears = case planet of
+      Mercury -> 0.2408467
+      Venus -> 0.61519726
+      Earth -> 1
+      Mars -> 1.8808158
+      Jupiter -> 11.862615
+      Saturn -> 29.447498
+      Uranus -> 84.016846
+      Neptune -> 164.79132
+```
+
+This approach uses a `case` expression to choose the relevant orbital period.
+Also a `where` clause is used to give names to important values.
+This tends to greatly improve readability.
+
+
+## General guidance
+
+### `where` clauses are your friend!
+
+Giving meaningful names to subexpressions can do wonders for code readability.
+`let` expressions allow the same.
+Being expressions &ndash; which `where` clauses aren't &ndash; `let` expressions are a bit more flexible in their use.
+However, `where` clauses list the local definitions _after_ the main expression.
+This allows you to paint the broad strokes of your strategy first, and to fill in the details later.
+This is so convenient that it amply compensates for `where` clauses not being expressions.
+
+More on `where` and `let` elsewhere:
+
+- Haskell Wiki: [Let vs. Where][haskellwiki-let-vs-where]
+- Haskell Wikibook:
+  - [`where` clauses][wikibook-where]
+  - [`let` bindings][wikibook-let]
+  - [`let` and `where` revisited][wikibook-let-vs-where]
+
+
+### `case` expressions are also your friend!
+
+Many beginning Haskellers write code like
+
+```haskell
+ageOn planet seconds
+  | planet == Mercury = _
+  | planet == Venus   = _
+  | {- etc. -}        = _
+```
+
+This is an [anti-pattern][wikipedia-anti-pattern].
+Pattern-match instead, for example using a `case` expression:
+
+```haskell
+ageOn planet seconds =
+  case planet of
+    Mercury    -> _
+    Venus      -> _
+    {- etc. -} -> _
+```
+
+Pattern matching is a fundamental concept in Haskell, but this document is too short to be able to fully explain it.
+Please consult your other learning resources.
+The track docs include an article on [Haskell learning resources][learning-resources].
+
+Pattern matching with `case` has benefits over using guards:
```

Comment:
Will it be readily apparent to a learner that "using guards" refers to the above solution that includes `| planet == "Mercury"`? If is already apparent, that was my only question about this pull request, so I think it is ready.

If it would not be apparent and needs to be made that way, mentioning guards somewhere around the sentence that reads "this is an anti-pattern" would make it clear.

Reply:
Good catch. I have changed

```diff
- This is an [anti-pattern][wikipedia-anti-pattern].
+ Using guards like this is an [anti-pattern][wikipedia-anti-pattern].
```